### PR TITLE
Add new landing page

### PR DESCRIPTION
**Pull request by DonOmalVindula**
## Purpose
 Implements a new landing page to on-prem documentation site.

## Goals
This PR replaces the existing landing page of the on-prem documentation site with a new design proposed in the issue mentioned below.

Resolves https://github.com/wso2/product-is/issues/11551

## Approach

1. A new template `landing-page.html` was created in `en/theme/material/templates/` as the base layout of the landing page.

2. `en/docs/index.md` file of the project was completely changed to accommodate the new design for the landing page.

3. In `en/theme/material/partials` directory, `header.html`, `tabs-item.html` and `footer.html` was changed to align the header and footer UI with the new design.

4. Custom files `search.html` and `source.html` were used  to customize the site search UI and Github repo links.

5. Added 9 new icons to the project in `en/docs/assets/img/icons/` from the new design

6. `en/docs/assets/css/theme.css` file was changed to accommodate new UI changes and mobile responsiveness.

Implmented Landing Page

### CODE REVIEW

File: en/docs/assets/css/theme.css [hunk 1]
```diff
@@ -1203,4 +1203,351 @@ html .md-footer-meta.md-typeset a,
...
+
+    .md-search__input {
+        background-color: white !important;
+    }
+}
 
+/* Overriding default font in internal pages on docs */
+html, body {
+    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif !important;
+}
```
Comment (mefarazath) [inline en/docs/assets/css/theme.css:1553]:
```suggestion
}

```

File: en/docs/assets/css/theme.css [hunk 2]
```diff
@@ -1203,4 +1203,351 @@ html .md-footer-meta.md-typeset a,
...
+    font-size: 20px !important;
+}
+
+.icon-card {
+    cursor: unset !important;
+    width: 4em !important;
+    height: 4em; /* 1:1 Aspect Ratio */
+    border: none !important;
+    box-shadow: 1px 6px 18px 4px rgba(77,77,77,0.15);
+    -webkit-box-shadow: 1px 6px 18px 4px rgba(77,77,77,0.15);
```
Comment (NipuniBhagya) [inline en/docs/assets/css/theme.css:1411]: Isn't setting the box-shadow attribute enough?

File: en/docs/assets/css/theme.css [hunk 3]
```diff
@@ -1203,4 +1203,351 @@ html .md-footer-meta.md-typeset a,
...
+/* Change color of dropdown links on hover */
+.scenario-dropdown-content a:hover {background-color: #ddd;}
+
+/* Show the dropdown menu on hover */
+.scenario-dropdown:hover .scenario-dropdown-content {display: block;}
+
+@media only screen and (min-width: 769px) {
+    [data-md-toggle=search]:checked~.md-header .md-search__inner {
+        width: 14rem;
+    }
```
Comment (NipuniBhagya) [inline en/docs/assets/css/theme.css:1535]: Please fix the formatting here.

File: en/docs/assets/img/icons/landing-page/admin.svg [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="23.999" viewBox="0 0 24 23.999">
```
Comment (NipuniBhagya) [inline en/docs/assets/img/icons/landing-page/admin.svg:1]: Add WSO2 license headers to all the newly added files.

File: en/docs/index.md [hunk 5]
```diff
@@ -1,189 +1,297 @@
...
+                        <h5 class="scenario-card-title">
+                           <i class="material-icons-outlined scenario-card-icon pe-1">
+                              <img src="../../assets/img/icons/landing-page/tick.svg">
+                           </i>
+                           <div class="row">
+                              Enable Single Sign On
+                           </div>
+                        </h5>
+                        <div class="ps-4">
+                           <p class="scenario-card-text">Allow users to securely log in to multiple applications by providing their credentials just once.</p>
```
Comment (NipuniBhagya) [inline en/docs/index.md:75]: Please fix the formatting here.

File: en/theme/material/partials/search.html [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        <div class="md-search-result" data-md-component="search-result">
+          <div class="md-search-result__meta">
+            {{ lang.t("search.result.initializer") }}
+          </div>
+          <ol class="md-search-result__list"></ol>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (NipuniBhagya) [inline en/theme/material/partials/search.html:44]: Please add a new line here.

File: en/theme/material/partials/source.html [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  class="md-source"
+  data-md-component="source"
+>
+  <div class="md-source__icon md-icon">
+    <i class="fa fa-github git-icon"></i>
+  </div>
+  <div class="md-source__repository">
+    {{ config.repo_name }}
+  </div>
+</a>
```
Comment (NipuniBhagya) [inline en/theme/material/partials/source.html:16]: Add a new line here as well.

File: en/theme/material/templates/landing-page.html [hunk 8]
```diff
@@ -0,0 +1,120 @@
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline en/theme/material/templates/landing-page.html:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: en/theme/material/templates/landing-page.html [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      text-decoration: none !important;
+    }
+    .nav-link-text {
+      color: white !important;
+    }
+    .md-tabs__link--active {
+      color: #fa6400 !important;
+    }
+    .md-main__inner {
+			max-width: unset !important;
```
Comment (NipuniBhagya) [inline en/theme/material/templates/landing-page.html:42]: Please fix the formatting here.

**Review by NipuniBhagya — CHANGES_REQUESTED** [review]

**Review by NipuniBhagya — APPROVED** [review]

**Pull request merged**